### PR TITLE
Log 10787 separate out retry driver

### PR DESCRIPTION
**Pull request by c-nixon**

### CODE REVIEW

File: common/http/src/retry.rs [hunk 1]
```diff
@@ -44,88 +55,169 @@ struct DiskRead {
...
+                if state.waiting.is_empty() {
+                    if let Err(e) = state.fill_waiting().await {
+                        return Some((Err(e), state));
+                    };
+                    // If there are still no objects then sleep for a while
+                    if state.waiting.is_empty() {
+                        Delay::new(Duration::from_secs(
+                            state.retry_base_delay_secs.try_into().unwrap_or(2),
+                        ))
+                        .await;
```
Comment (jorgebay) [inline common/http/src/retry.rs:125]: should we `continue` here?

File: common/http/src/retry.rs [hunk 2]
```diff
@@ -44,88 +55,169 @@ struct DiskRead {
...
+                    }
+                }
+
+                if let Some(path) = state.waiting.pop() {
+                    match Retry::read_from_disk(&path).await {
+                        Ok((offsets, ingest_body)) => {
+                            match IntoIngestBodyBuffer::into(ingest_body).await {
+                                Ok(body_buffer) => {
+                                    return Some((Ok((body_buffer, offsets)), state))
```
Comment (jorgebay) [inline common/http/src/retry.rs:134]: Should we throttle, e.g. in the order of 50-100ms, after a successful read to prioritize new logs?

**Review by jorgebay — COMMENTED** [review]
New retry logic looks great, having them in a separate stream is a good call.

File: common/http/src/retry.rs (hunk 2 above)
Comment (c-nixon) [reply common/http/src/retry.rs]: We could reintroduce the retry step delay config?

I accidentally pushed to the wrong branch so I've reopened the PR over at https://github.com/logdna/logdna-agent-v2/pull/222

**Pull request closed without merging**